### ISSUE TITLE
Some deprecation warning on a fresh new install: is the project maintaned? 

### ISSUE THREAD

**Issue by andreamoro** [lektor/lektor#949]
Approaching Lektor right now, so haven't had a chance to investigate too much into the code, but a plain lektor server on an unbuntu 21.10 with Flash 2.0.2 produces the following warning.

```
/home/andrea/GitHub/websites/andreamoro-ssg-lektor/lib/python3.9/site-packages/jinja2/environment.py:362: DeprecationWarning: The 'autoescape' extension is deprecated and will be removed in Jinja 3.1. This is built in now.
  self.extensions = load_extensions(self, extensions)
/home/andrea/GitHub/websites/andreamoro-ssg-lektor/lib/python3.9/site-packages/jinja2/environment.py:362: DeprecationWarning: The 'with' extension is deprecated and will be removed in Jinja 3.1. This is built in now.
  self.extensions = load_extensions(self, extensions)
/home/andrea/GitHub/websites/andreamoro-ssg-lektor/lib/python3.9/site-packages/lektor/environment.py:455: DeprecationWarning: 'contextfilter' is renamed to 'pass_context', the old name will be removed in Jinja 3.1.
  url=jinja2.contextfilter(lambda ctx, *a, **kw: url_to(*a, **kw)),
/home/andrea/GitHub/websites/andreamoro-ssg-lektor/lib/python3.9/site-packages/lektor/environment.py:456: DeprecationWarning: 'contextfilter' is renamed to 'pass_context', the old name will be removed in Jinja 3.1.
  asseturl=jinja2.contextfilter(
/home/andrea/GitHub/websites/andreamoro-ssg-lektor/lib/python3.9/site-packages/lektor/environment.py:459: DeprecationWarning: 'contextfilter' is renamed to 'pass_context', the old name will be removed in Jinja 3.1.

```

Therefore my question is simple. Is this project stable and currently maintained to a point that I can invest some time to migrate my current site into this platform?

**Comment by dairiki** [comment]
How are you installing lektor? I suspect that you are dealing with an older version of lektor.  The latest release version is 3.2.2 — any other version is well over a year old.

----

There are a number of us who use lektor for production sites.  The project is certainly *stable* in that sense.

From my perspective, there has been a lot of churn in the pool of core contributors and maintainers over the years. I believe many of those who started the project are only tangentially involved these days.  That said, bugs do get fixed.  I, for one, would like to see more frequent releases.  (E.g at present, the master branch has diverged a bit from the released branch.)

As I understand, Lektor was conceived as a project to allow the developers (not so computer-savvy) parents edit and maintain the website for their business.  In its current state, I don't think it is probably the best choice for that use case.
That said, if you have some python skills and/or lektor offers the features that you need, IMO there is no reason not to use it in a new project.  (I would.)

**Comment by andreamoro** [comment]
Hi @dairiki

I've installed Lektor via pip as suggested on their home page. And in fact I can see the v. 3.2.2 (the same I can see on git??) installed along with Flask 2.0.2. The two should be compatible as far as I can tell, so I'm not sure why the exception is raised.

In which way the master master has diverged from the release? Master in theory should be the one released. Am I missing something here?

As for the last part of your message. 
You might be surprised, but other frameworks I recently tested these days like Pelican and Nikola both fail in simple things like localisation or generating independent output in separate folders. Something I was after because I have my site in two languages, but content is rarely specular although the same localised template is used. 
So in my previous setup, with Middleman (using Ruby), I did manage to have one big repository and control where the content was deployed via the file extension (e.g. .en.html)

I could achieve /part of the above Lektor almost immediately. The only annoying bits so far are:
- using alternative, content become specular in all alternatives, and I can only disable a certain output by enabling/disabling content via the `.lr`. Is there any other way? 
- the static folder is common between all alternatives, is there any way to make it inside the alternative folders?
- I could not make the i18n plugin to work to localise the template. When running the `lektor server` I get the server entering in an error loop.

I have a decent Python knowledge, though limited time to look properly into the source code to hack it, if you know what I mean. I rather prefer looking into long term solutions. So I'm not afraid to support the community if something can be done better, but equally I would love some support to avoid investigations that many may already have done?

**Comment by andoresuperesu** [comment]
@andreamoro you might find this as an introduction to the current state of things:

https://github.com/lektor/lektor/pull/929

**Comment by andreamoro** [comment]
@andresperezcera thanks. Not sure I was able to find anything useful, but it sounds like contributions are regular.

I've just decided to make it a go in using, and perhaps attempting to contribute, but I've certainly my own limits. So, may I ask what's the best way to debug you guys use? 
A simple run in VsCode doesn't work due to the relative imports.

**Comment by dairiki** [comment]
@andreamoro The deprecation warnings you reported were fixed in #921, however, you are right in that #921 has not made it into the 3.2.x releases.

Hopefully, we can get the master branch released soon.  I'm not at all sure what it is that's holding that up.

**Comment by dairiki** [comment]
This is fixed in the 3.3.0 release.
